SimG4CMS/Calo/src/CaloTrkProcessing.cc

```diff
@@ -186,6 +183,53 @@ void CaloTrkProcessing::update(const G4Step* aStep) {
     throw cms::Exception("Unknown", "CaloTrkProcessing") << "cannot get trkInfo for Track " << id << "\n";
   }
 
+  if (doFineCalo_) {
+    // Boundary-crossing logic
+    int prestepLV = isItCalo(aStep->GetPreStepPoint()->GetTouchable(), fineDetectors_);
+    int poststepLV = isItCalo(aStep->GetPostStepPoint()->GetTouchable(), fineDetectors_);
+    if (prestepLV < 0 && poststepLV >= 0
+        // Allow back-scattering and filter it out later; ensure consistency during the SIM step
+        // && std::abs(theTrack->GetStep()->GetPreStepPoint()->GetPosition().z()) < std::abs(theTrack->GetPosition().z())
+    ) {
+      edm::LogVerbatim("DoFineCalo") << "Entered fine volume " << poststepLV << ":"
```

Comment:
This LogVerbatim should be protected by verbosity flag or be within #ifdef

Reply:
I've added an ifdef